ddtrace/contrib/django/patch.py

```diff
@@ -351,91 +351,76 @@ def traced_template_render(django, pin, wrapped, instance, args, kwargs):
+                op_name = "django.view.{0}".format(name)
+                trace_utils.wrap(cls, name, traced_func(django, name=op_name, resource=resource))
             except Exception:
-                log.debug("Failed to instrument Django response %r function %s", response_cls, name, exc_info=True)
+                log.debug("Failed to instrument Django view %r function %s", cls, name, exc_info=True)
+
+        # Patch response methods
+        response_cls = getattr(cls, "response_class", None)
+        if response_cls:
+            methods = ("render",)
+            for name in methods:
+                try:
+                    func = getattr(response_cls, name, None)
+                    # Do not wrap if the method does not exist or is already wrapped
+                    if not func or isinstance(func, wrapt.ObjectProxy):
+                        continue
+
+                    resource = "{0}.{1}".format(func_name(response_cls), name)
+                    op_name = "django.response.{0}".format(name)
+                    trace_utils.wrap(response_cls, name, traced_func(django, name=op_name, resource=resource))
+                except Exception:
+                    log.debug("Failed to instrument Django response %r function %s", response_cls, name, exc_info=True)
 
+
+def _instrument_view(django, view):
+    """Helper to wrap Django views."""
     # If the view itself is not wrapped, wrap it
-    if not isinstance(view, wrapt.ObjectProxy):
+    if callable(view) and not isinstance(view, wrapt.ObjectProxy):
+        if hasattr(view, "view_class"):
+            # Accessing the name of a class based view using view.__class__ is not supported in Django 4.0
+            # view_class should be used instead
+            view_name = func_name(view.view_class)
+        else:
+            view_name = func_name(view)
+
         view = wrapt.FunctionWrapper(
-            view, traced_func(django, "django.view", resource=func_name(view), ignored_excs=[django.http.Http404])
+            view, traced_func(django, "django.view", resource=view_name, ignored_excs=[django.http.Http404])
         )
     return view
 
 
 @trace_utils.with_traced_module
-def traced_urls_path(django, pin, wrapped, instance, args, kwargs):
-    """Wrapper for url path helpers to ensure all views registered as urls are traced."""
-    try:
-        if "view" in kwargs:
-            kwargs["view"] = instrument_view(django, kwargs["view"])
-        elif len(args) >= 2:
-            args = list(args)
-            args[1] = instrument_view(django, args[1])
-            args = tuple(args)
-    except Exception:
-        log.debug("Failed to instrument Django url path %r %r", args, kwargs, exc_info=True)
-    return wrapped(*args, **kwargs)
-
+def traced_resolve_url_pattern(django, pin, wrapped, instance, args, kwargs):
+    """Django wrapper to ensure all resolved views are traced"""
+    resolver_match = wrapped(*args, **kwargs)
+    if resolver_match is None:
+        return
```

Comment:
It might be better to change this to:

```python
resolver_match = wrapped(*args, **kwargs)
if resolver_match:
    # wrap the result
return resolver_match
```

That is, we should always return `resolver_match`.

Reply:
Yup that looks cleaner